Feature/current plan endpoint

DerDhaval opened the pull request:
What: 

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

princeparmar committed:
Add plan_id column to billing_transactions table

- Introduced a new column `plan_id` to the `billing_transactions` table to associate transactions with specific plans.
- Updated the database migration script to include the new column and ensure proper handling in SQL queries.
- Modified relevant data structures and methods to accommodate the new `plan_id` field, enhancing transaction data management.

DerDhaval committed:
Add new billing and dashboard functionalities

- Implemented `GetBillingTransactions` method in the Service to retrieve billing transactions for users, enhancing billing data access.
- Added `GetAPIKeysStore` method to provide access to the API keys store interface.
- Updated server routing to include a new endpoint for dashboard statistics, improving user access to dashboard data.
- Introduced new data structures for dashboard cards, including `AutoSyncCard`, `VaultCard`, `AccessCard`, and `BillingCard`, to support the dashboard UI.
- Created `GetHardcodedDashboardCards` function to return sample dashboard data for UI development, facilitating frontend integration.

DerDhaval committed:
Add dashboard card configuration and enhance billing transaction handling

- Introduced `ConfigTypeDashboardCards` to support dashboard card configurations.
- Updated `Config` struct to include `BackupToolsURL` for AutoSync stats integration.
- Enhanced `NewAuth` constructor to accept `BackupToolsURL` parameter.
- Modified `CreateTokenPaymentBillingTransaction` to accept an optional `planID` for billing transactions.
- Updated database handling to include `planID` in billing transactions, improving transaction data management.
- Implemented dynamic data enrichment for dashboard cards, including billing, vault, access, and AutoSync statistics.

princeparmar commented on satellite/console/consoleweb/consoleapi/auth.go:
```diff
@@ -95,7 +99,7 @@ type UserDetails struct {
 
 // NewAuth is a constructor for api auth controller.
-func NewAuth(log *zap.Logger, service *console.Service, accountFreezeService *console.AccountFreezeService, mailService *mailservice.Service, cookieAuth *consolewebauth.CookieAuth, analytics *analytics.Service, satelliteName, externalAddress, letUsKnowURL, termsAndConditionsURL, contactInfoURL, generalRequestURL string, activationCodeEnabled bool, badPasswords map[string]struct{}) *Auth {
+func NewAuth(log *zap.Logger, service *console.Service, accountFreezeService *console.AccountFreezeService, mailService *mailservice.Service, cookieAuth *consolewebauth.CookieAuth, analytics *analytics.Service, satelliteName, externalAddress, letUsKnowURL, termsAndConditionsURL, contactInfoURL, generalRequestURL string, activationCodeEnabled bool, badPasswords map[string]struct{}, backupToolsURL  ...
```
remove logic from auth and add in service

DerDhaval committed:
Enhance dashboard functionality with Backup Tools integration

- Added `BackupToolsURL` parameter to the `NewService` and `NewAPI` constructors for improved integration with Backup Tools.
- Updated `GetDashboardStats` method to enrich AutoSync card data using Backup Tools statistics.
- Refactored `Auth` controller to streamline dashboard data retrieval and response handling.
- Removed redundant parameters and methods related to Backup Tools from the `Auth` controller, simplifying the codebase.

DerDhaval committed:
Refactor dashboard statistics handling and introduce new API endpoint

- Moved `GetDashboardStats` method from the `Auth` controller to a new `Dashboard` controller for better separation of concerns.
- Implemented a new API endpoint for dashboard statistics under `/api/v0/dashboard/stats`, enhancing user access to dashboard data.
- Updated server routing to include the new dashboard statistics endpoint, improving overall API structure and maintainability.

princeparmar commented on satellite/console/service.go:
```diff
@@ -6242,3 +6255,298 @@ func (s *Service) RevokeUserDeveloperAccess(ctx context.Context, clientID string
+	ActiveSyncsLabel string   `json:"activeSyncsLabel"`
+	FailedSyncs      int      `json:"failedSyncs"`
+	FailedSyncsLabel string   `json:"failedSyncsLabel"`
+}
```
type Card struct {
	Title            string   `json:"title"`
	Description      string   `json:"description"`
	Icon             CardIcon `json:"icon"`
	Status           string   `json:"status"`
	ButtonLink       *string  `json:"buttonLink"`
}


type AutoSyncCard struct {
Card
	ActiveSyncs      int      `json:"activeSyncs"`
	ActiveSyncsLabel string   `json:"activeSyncsLabel"`
	FailedSyncs      int      `json:"failedSyncs"`
	FailedSyncsLabel string   `json:"failedSyncsLabel"`
}

princeparmar commented on satellite/console/service.go:
```diff
@@ -6242,3 +6255,298 @@ func (s *Service) RevokeUserDeveloperAccess(ctx context.Context, clientID string
+	}
+
+	card.Status = "active"
+	transactions, err := s.GetBillingTransactions(ctx, user.ID)
```
create new query for this

princeparmar commented on satellite/console/service.go:
```diff
@@ -6242,3 +6255,298 @@ func (s *Service) RevokeUserDeveloperAccess(ctx context.Context, clientID string
+}
+
+// convertValidityToDuration converts plan validity to time.Duration
+func (s *Service) convertValidityToDuration(plan *billing.PaymentPlans) time.Duration {
```
check if we can use value from project table

princeparmar commented on satellite/console/service.go:
```diff
@@ -6242,3 +6255,298 @@ func (s *Service) RevokeUserDeveloperAccess(ctx context.Context, clientID string
+	}); err == nil {
+		card.ActiveUsers = int(members.TotalCount) + 1 // + owner
+		card.AdminUsers = 1
+	}
```
return count of access we have created

DerDhaval committed:
Refactor dashboard card structures and enhance billing transaction retrieval

- Renamed `CardIcon` to `Icon` and introduced a new `BaseCard` struct to streamline dashboard card definitions.
- Updated `AutoSyncCard`, `VaultCard`, `AccessCard`, and `BillingCard` to inherit from `BaseCard`, reducing redundancy.
- Added `GetLatestCompletedDebitTransaction` method to the billing transactions interface for improved transaction retrieval.
- Implemented database query for fetching the latest completed debit transaction, enhancing billing data management.
